rc/src/lib.rs

```diff
@@ -0,0 +1,62 @@
+use std::ops::Deref;
+use std::mem;
+
+pub struct MyRc<T> {
+    value: *mut T,
+    pub r_count: *mut usize,
```

Comment:
Why is this `pub`? That allows the user to overwrite the reference count, causing memory errors.